src/lib/viewers/controls/controls-layer/__tests__/ControlsLayer-test.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mount, ReactWrapper } from 'enzyme';
+import ControlsLayer, { HIDE_CLASSNAME, HIDE_DELAY_MS } from '../ControlsLayer';
+
+describe('ControlsLayer', () => {
+    const children = <div className="TestControls">Controls</div>;
+    const getElement = (wrapper: ReactWrapper): ReactWrapper => wrapper.childAt(0);
+    const getWrapper = (props = {}): ReactWrapper => mount(<ControlsLayer {...props}>{children}</ControlsLayer>);
+
+    beforeEach(() => {
+        jest.useFakeTimers();
+    });
+
+    describe('event handlers', () => {
+        test.each(['focus', 'mouseenter'])('should show the controls %s', eventProp => {
+            const wrapper = getWrapper();
+
+            act(() => {
+                getElement(wrapper).simulate(eventProp);
+            });
+            wrapper.update();
+
+            expect(getElement(wrapper).hasClass(HIDE_CLASSNAME)).toBe(true);
+        });
+
+        test.each`
+            showTrigger     | hideTrigger
+            ${'focus'}      | ${'blur'}
+            ${'mouseenter'} | ${'mouseleave'}
+        `('should show $showTrigger and hide $hideTrigger', ({ hideTrigger, showTrigger }) => {
+            const wrapper = getWrapper();
+
+            act(() => {
+                getElement(wrapper).simulate(showTrigger);
+            });
+            wrapper.update();
+
+            expect(getElement(wrapper).hasClass(HIDE_CLASSNAME)).toBe(true);
```

Comment:
Should these checks be reversed? After show trigger shouldn't  HIDE_CLASSNAME not be present?

Reply:
Oh man, good catch. The test is right, but I inverted the name of the variable somehow; it should be `SHOW_CLASSNAME`.